src/net/encryption/src/key.rs

```diff
@@ -0,0 +1,42 @@
+use rsa::{RsaPublicKey, RsaPrivateKey, pkcs1::EncodeRsaPublicKey, Pkcs1v15Encrypt};
+use rand::rngs::OsRng;
+
+pub struct KeyPair {
+    pub public_key: RsaPublicKey,
+    pub encoded_public_key: der::Document,
+    pub private_key: RsaPrivateKey,
+}
+
+impl KeyPair {
+    pub fn new() -> Self {
+        let mut rng = OsRng;
+        let private_key = RsaPrivateKey::new(&mut rng, 1024).expect("Failed to generate a key");
+        let public_key = RsaPublicKey::from(&private_key);
+        let encoded_public_key = public_key.to_pkcs1_der().expect("Failed to encode public key");
+
+        Self {
+            public_key,
+            encoded_public_key,
+            private_key,
+        }
+    }
+
+    pub fn encrypt (&self, data: &Vec<u8>) -> Vec<u8> {
+        let mut rng = OsRng;
+        self.public_key.encrypt(&mut rng, Pkcs1v15Encrypt, &data).expect("Failed to encrypt data")
```

Comment:
Maybe for error handling, add error variants to `errors.rs` and instead of `.expect()`, use `.map_err()?` or `.ok_or()?` ?

Other than that, everything looks good and i agree with Asurar0's comments